docs/tutorials/controlling-wedo2-motor.md

```diff
@@ -0,0 +1,235 @@
+version and check for the existence of the bluez package:
+
+    robot@ev3dev:~# sudo apt-get install bluez
+    Reading package lists... Done
+    Building dependency tree       
+    Reading state information... Done
+    bluez is already the newest version.
+
+For shell scripts we need one command from bluez - the gatttool command:
+
+    robot@ev3dev:~# gatttool
+    Usage:
+      gatttool [OPTION...]
+    ...
+
+
+Now we need to find the bluetooth address of our WeDO 2.0 hub.
+For that we press it's button to put it in descovery mode and run this
+command:
+
+    robot@ev3dev:~# sudo hcitool -i hci0 lescan
+    LE Scan ...
+    A0:E6:F8:1E:58:57 (unknown)
+    A0:E6:F8:1E:58:57 	
+
+In the example above, `A0:E6:F8:1E:58:57` is the bluetooth address of our WeDO 2.0
+hub. We can use other tools, including a smartphone with BLE support - the WeDO 2.0
+will probably show up as `LPF2 Smart Hub 2 I/O`, just look for the address in its
+properties.
+
+
+## Shell script example
+
+This short shell script makes a motor connected to the first port spin for one second
+then stop:
+
+    #!/bin/bash
+    gatttool -i hci0 -b A0:E6:F8:1E:58:57 --char-write -a 0x003d -n 01010164
+    sleep 1
+    gatttool -i hci0 -b A0:E6:F8:1E:58:57 --char-write -a 0x003d -n 01010100
+
+(You need to run this script with sudo, unless you already have root previleges)
+
+We see that it uses the gatttool command to send a sequence of 4 bytes to one specific
+handler (0x003d). The WeDO 2.0 has several handlers but until LEGO Education releases
+the promised SDK this is the only handler we «know» how to use:
+
+This is meaning of those 4 bytes:
+* the firt byte defines the port (01 or 02)
+* the second byte defines the command (01 = motor speed)
+* the third byte defines the lenght of the following argument(s) (01)
+* the fouth byte is the argument, in this case the speed percentage
+
+To spin in one direction we send a positive value from 1 to 100 (or 01 to 64 in
+hexadecimal).
+To spin in the opposite direction we send a «negative» value from
+255 to 156 (or FF to 9C in hexadecimal).
+To stop the motor we set the speed as zero (00).
+Please note that for small speed values (less than 20%) the motor will not respond.
+
+
+## Python example
+
+To use pyhton with the WeDO 2.0 we need a BLE library. Unfortunately BLE
+support in python is still quite imature but at least pybluez offers an extension
+for BLE:
+
+    pip install pybluez
+    pip install pybluez[ble]
+
+This didn't work in my EV3 but since the ble extension is in fact gattlib I went for
+it:
+
+    sudo apt-get install pkg-config libboost-python-dev libboost-thread-dev \
+     libbluetooth-dev libglib2.0-dev python-dev
+
+    pip install gattlib
+
+Please note that it takes A LOT of memory and around 2 hours time to install gattlib.
+After some failures ("virtual memory exhausted: Cannot allocate memory") I finally
+succeeded extending my ev3dev swapfile to 1024 MB.
```

Comment:
i did iby editing "/etc/dphys-swapfile, just changing this line:
CONF_SWAPSIZE=1024

Then restarting dphys-swapfile:
/etc/init.d/dphys-swapfile stop
/etc/init.d/dphys-swapfile start

but since this is an ev3dev system that I've been updating almost since 2 years ago I'm not sure if dphys-swapfile is still used with latest image?